pkgs/modules/python/pip.nix

```diff
@@ -18,6 +18,41 @@ let
     ];
   });
 
+  # This pip wrapper will detect if python is called from within
+  # the virtualenv: `.pythonlibs/bin`. If so, it will install a shell script
+  # called `pip` into the virtualenv so that if pip is called in the future
+  # python will think it's in the virtualenv.
+  pip-wrapper = pkgs.writeShellApplication {
+    name = "pip";
+    text = ''
+      python_location=$(which python)
+      venv_pip_location="''${REPL_HOME}/.pythonlibs/bin/pip"
+      venv_pip3_location="''${REPL_HOME}/.pythonlibs/bin/pip3"
+
+      function init_venv_pip {
+        cat <<EOF > "''${venv_pip_location}"
+      #! /bin/sh
+      export PIP_CONFIG_FILE=
+      python -m pip "\$@"
+      EOF
+        chmod u+x "''${venv_pip_location}"
+        if [ ! -f "''${venv_pip3_location}" ]; then
+          ln -s "''${venv_pip_location}" "''${venv_pip3_location}"
+        fi
+      }
+
+      if [ "$python_location" = "''${REPL_HOME}/.pythonlibs/bin/python" ]; then
+        if [ ! -f "''${venv_pip_location}" ]; then
+          init_venv_pip
+        fi
+        exec "''${venv_pip_location}" "$@"
+      else
+        unset PYTHONNOUSERSITE
+        exec "${pip}/bin/pip"  "$@" 
+      fi
```

Comment:
Do we want to preserve this legacy path?

Reply:
Yes. It is used when we are not using uv/virtualenv and still using poetry.

Reply:
ack